AndroidAnnotations/androidannotations-rest-spring/rest-spring/src/main/java/org/androidannotations/rest/spring/helper/RestAnnotationHelper.java

```diff
@@ -295,6 +301,29 @@ public JExpression getResponseClass(Element element, RestHolder holder) {
 		return responseClassExpr;
 	}
 
+	private boolean checkIfParameterizedTypeReferenceShouldBeUsed(TypeMirror returnType) {
+		switch (returnType.getKind()) {
+		case DECLARED:
+			return !((DeclaredType) returnType).getTypeArguments().isEmpty();
+
+		case ARRAY:
+			ArrayType arrayType = (ArrayType) returnType;
+			TypeMirror componentType = arrayType.getComponentType();
+			if (componentType.getKind() == TypeKind.DECLARED) {
+				return !((DeclaredType) componentType).getTypeArguments().isEmpty();
+			} else {
+				return checkIfParameterizedTypeReferenceShouldBeUsed(componentType);
+			}
+		}
+		return false;
+	}
+
+	public JExpression createParameterizedTypeReferenceAnonymousSubclassInstance(TypeMirror returnType, RestHolder holder) {
+		JClass narrowedTypeReference = holder.getEnvironment().getJClass(RestSpringClasses.PARAMETERIZED_TYPE_REFERENCE).narrow(codeModelHelper.typeMirrorToJClass(returnType));
```

Comment:
You can call `getEnvironment` directly instead of `holder.getEnvironment()`.
This way, you don't need to give the `holder` as a parameter of the method.